### PR TITLE
Initial PPC64LE support

### PR DESCRIPTION
**Pull request by johncurley**
These are the file changes for LuaJIT PPC64le from the https://github.com/LuaJIT/LuaJIT/pull/54/files repo. It's the first component of PPC64le support for OXR though I don't think JIT is working at this point. However it has been tested and runs on PPC64le systems so better to have it than not.

### CODE REVIEW

File: src/lj_ccall.c [hunk 1]
```diff
@@ -419,6 +444,8 @@
     CCALL_HANDLE_GPR \
   }
 #endif
+#endif /* LJ_ABI_SOFTFP */
```
Comment (Xottab-DUTY) [inline src/lj_ccall.c:446]: This is a mistakenly added endif
```suggestion
```

Comment (johncurley) [reply src/lj_ccall.c:446]: Looks like I've got the comment wrong. The comment should be on the line above but the #endif is okay I think.

Comment (johncurley) [reply src/lj_ccall.c:446]: Actually that is an extra #endif so can be safely removed. Hopefully it compiles after this.

File: src/lj_ccall.c [hunk 2]
```diff
@@ -419,6 +444,8 @@
     CCALL_HANDLE_GPR \
   }
 #endif
```
Comment (Xottab-DUTY) [inline src/lj_ccall.c]:
```suggestion
```

File: src/lj_ccall.c (hunk 1 above)
Comment (Xottab-DUTY) [reply src/lj_ccall.c:446]: Thank you!

File: dynasm/dasm_ppc.lua [hunk 3]
```diff
@@ -1718,7 +1720,12 @@ op_template = function(params, template, nparams)
     elseif p == "G" then
       op = op + parse_imm(params[n], 8, 12, 0, false); n = n + 1
     elseif p == "H" then
-      op = op + parse_shiftmask(params[n], true); n = n + 1
+	  v = parse_imm(params[n], 6, 0, 0, false);
+      op = op + shl(band(v,31), 11)+shl(shr(v,5), 1);
+      n = n + 1;
+    elseif p == "f" then
+      v = tonumber(params[n]);
+      op = op + shl(band(v,31), 11)+shl(shr(v,5), 1);
```
Comment (Xottab-DUTY) [inline dynasm/dasm_ppc.lua:1728]: I'm a bit worried because this changes both PPC and PPC64, but, hopefully, it's ok.

**Review by Xottab-DUTY — APPROVED** [review]

File: src/lj_arch.h [hunk 4]
```diff
@@ -316,6 +316,10 @@
 #error "No support for PPC64"
 #endif
 
+#if _ARCH_PWR9
+#define LJ_ARCH_VERSION		90
+#if _ARCH_PWR8
```
Comment (Xottab-DUTY) [inline src/lj_arch.h:321]: It should be `#elif` but I missed it and merged :)

Comment (Xottab-DUTY) [reply src/lj_arch.h:321]: Committed a fix: 0bc04c4768a71a6bf7f7d6a973b06a97e58dc722

**Pull request merged**